containers/Signup/Signup.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import { ScrollView, StatusBar, Text, TextInput, TouchableOpacity, View } from "react-native";
+import { styles } from "./style";
+
+export const Signup = ({ navigation }: { navigation: any }) => {
+
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [isError, setIsError] = useState(false);
+    const [emailErrorMessage, setEmailErrorMessage] = useState("");
+    const [isPassError, setIsPassError] = useState(false);
+    const [passErrorMessage, setPassErrorMessage] = useState("");
+    
+    const handleSignupPress = () => {
+        navigation.navigate({
+            name: "Dashboard",
+            params: {
+                userName: email
+            }
+        });
+    }
+    const handleLoginPress = () => {
+        navigation.navigate("Login");
+    }
+
+    const validateSymbols = (email: string) => {
+        let re = /[ `!#$%^&*()_+\-=\[\]{};':"\\|,<>\/?~]/;
+          return re.test(email);
+    };
+    const validateAtss = (email: string) => {
+        let count = 0;
+        for(let i=0; i<email.length; i++){
+            if(email.charAt(i).includes("@"))
+                count++;
+        }
+        return count;
+    };
+    const validatePass = (password: string) => {
+        let message = "";
+        if(password.length < 8) message = "* Minimum length is 8 \n"
+        let check1 = /[0-9a-zA-z+!\/]/;
+        if(!check1.test(password)) message += "* Only letters, numbers and symbols are required \n"
+        else {
+            let check2 = /[0-9]/
+            if(!check2.test(password)) message += "* Atleast 1 letter is required \n"
+
+            let check3 = /[+!\/]/
+            if(!check3.test(password)) message += "* Atleast 1 symbol is required \n"
+
+        }
+        return message;
```

Comment:
Although this validates to some degree, I am still able to proceed with signing up.

I would have also liked to see more in the email validation, e.g. `must end in @___[domain]`

Reply:
True. It only validates to some extent, because I strictly followed the requirements mentioned in ReadMe.